src/libs/auth.js

```diff
@@ -238,6 +239,79 @@ authStore.subscribe((state, oldState) => {
   }
 })
 
+authStore.subscribe(withErrorReporting('Error loading Framework Services account status', async (state, oldState) => {
+  const loadFenceStatus = async () => {
+    try {
+      const dcpStatus = await Ajax().User.getFenceStatus('fence')
+      const dcfStatus = await Ajax().User.getFenceStatus('dcf-fence')
+
+      return { dcp: dcpStatus, dcf: dcfStatus }
+    } catch (error) {
+      if (error.status === 404) {
+        return {}
+      } else {
+        throw error
+      }
+    }
+  }
+
+  if (oldState.registrationStatus !== 'registered' && state.registrationStatus === 'registered') {
+    const fenceStatus = await loadFenceStatus()
+    authStore.update(state => ({ ...state, fenceDCFStatus: fenceStatus.dcp, fenceDCPStatus: fenceStatus.dcf }))
+  }
+}))
+
+//DCP
+authStore.subscribe(async (state, oldState) => {
+  if (state.fenceDCPStatus !== oldState.fenceDCPStatus) {
+    const notificationId = 'fence-dcp-link-warning'
+    const now = Date.now()
+    const expireTime = state.fenceDCPStatus && addDays(30, parseJSON(state.fenceDCPStatus.issued_at))
+    const expireStatus = Utils.cond(
+      [!expireTime, () => null],
+      [now >= expireTime, () => 'has expired'],
+      [now > expireTime - (1000 * 60 * 60 * 24 * 5), () => 'will expire soon']
+    )
+    const redirectUrl = `${window.location.origin}/${Nav.getLink('fence-callback')}`
+    const href = await Ajax().User.getFenceAuthUrl('fence', redirectUrl)
+    if (expireStatus) {
+      notify('info', div({}, [
+        `Your access to DCP Framework Services ${expireStatus}. To regain access, log-in to Framework Services to `,
+        [FrameworkServicesLink('re-link', href.url)],
+        ' your account'
+      ]), { id: notificationId })
+    } else {
+      clearNotification(notificationId)
+    }
+  }
+})
+
+//DCF
+authStore.subscribe(async (state, oldState) => {
+  if (state.fenceDCFStatus !== oldState.fenceDCFStatus) {
```

Comment:
There's a fair bit of similarity between these two subscriptions, so we could extract out a helper with the common code. There *might* be a case for trying to generalize over the NIH subscription as well, though that's less of a sure thing. It depends on how much we intend to unify those experiences.